src/x509.c

```diff
@@ -4786,41 +4791,51 @@ unsigned long wolfSSL_X509_subject_name_hash(const WOLFSSL_X509* x509)
     unsigned char*     canonName = NULL;
     byte               digest[WC_MAX_DIGEST_SIZE];
     int                size = 0;
+    int                retHash = NOT_COMPILED_IN;
 
     if (x509 == NULL) {
+        WOLFSSL_ERROR_MSG("WOLFSSL_X509 pointer was NULL");
         return 0;
     }
 
     subjectName = wolfSSL_X509_get_subject_name((WOLFSSL_X509*)x509);
 
     if (subjectName == NULL) {
-        WOLFSSL_MSG("wolfSSL_X509_get_subject_name error");
+        WOLFSSL_ERROR_MSG("wolfSSL_X509_get_subject_name error");
         return 0;
     }
 
     size = wolfSSL_i2d_X509_NAME_canon(subjectName, &canonName);
 
     if (size <= 0 || canonName == NULL){
-        WOLFSSL_MSG("wolfSSL_i2d_X509_NAME_canon error");
+        WOLFSSL_ERROR_MSG("wolfSSL_i2d_X509_NAME_canon error");
         return 0;
     }
 
     #ifndef NO_SHA
-    if (wc_ShaHash((const byte*)canonName, (word32)size, digest) != 0) {
-        WOLFSSL_MSG("wc_ShaHash error");
-        return 0;
-    }
+    retHash = wc_ShaHash((const byte*)canonName, (word32)size, digest);
     #elif !defined(NO_SHA256)
-    if (wc_Sha256Hash((const byte*)canonName, (word32)size, digest) != 0) {
-        WOLFSSL_MSG("wc_Sha256Hash error");
-        return 0;
-    }
+    retHash = wc_Sha256Hash((const byte*)canonName, (word32)size, digest);
+    #else
+    retHash = NOT_COMPILED_IN;
     #endif
 
-    hash  =  (unsigned long) digest[0];
-    hash |= ((unsigned long) digest[1]) << 8;
-    hash |= ((unsigned long) digest[2]) << 16;
-    hash |= ((unsigned long) digest[3]) << 24;
+    if (retHash == 0) {
+        /* 4 bytes in little endian as unsigned long */
+        hash = (((unsigned long)digest[3] << 24) |
+                ((unsigned long)digest[2] << 16) |
+                ((unsigned long)digest[1] <<  8) |
+                ((unsigned long)digest[0]));
+    }
+    else {
+        #ifndef NO_SHA
+        WOLFSSL_ERROR_MSG("wc_ShaHash error");
+        #elif !defined(NO_SHA256)
+        WOLFSSL_ERROR_MSG("wc_Sha256Hash error");
+        #else
+        WOLFSSL_ERROR_MSG("wolfSSL_X509_subject_name_hash not compiled in");
```

Comment:
Not totally accurate. There's no hash function compiled in.

Reply:
Had also thought about just returning NAME_hash() if using SHA-1 because in that case the code is identical.